src/elements/call-to-action.html

```diff
@@ -9,45 +9,47 @@
 
   <template>
 
-    <style include="shared-styles">
-      :host {
-        display: block;
-        background: var(--default-primary-color);
-        color: var(--text-primary-color);
-      }
-
-      .container {
-        @apply(--layout-vertical);
-        @apply(--layout-center-center);
-      }
-
-      .heading {
-        @apply(--layout-flex-auto);
-        @apply(--layout-self-center);
-        margin-bottom: 0 !important;
-      }
-
-      @media (max-width: 599px) {
-        .open-dialog-button {
-          margin-top: 10px;
+    {% if callToAction %}
```

Comment:
I think this kind of checking should be in parent component which includes this one.
The component itself shouldn't know about that. It just does its job - render cta, render videos
In parent component which encapsulates this one has to be

  {% if callToAction %}
<call-to-action></<call-to-action>
{% endif %}

So, in  that case we won't init that component at all
This suggestion applies to all the changes you've made. Please move all the condition upper, to the parent component. 
And thank you for the contribution, it's a good idea to have such configuration!